library/ssl_misc.h

```diff
@@ -2800,24 +2800,30 @@ static inline unsigned int mbedtls_ssl_session_get_ticket_flags(
            (flags & MBEDTLS_SSL_TLS1_3_TICKET_FLAGS_MASK);
 }
 
-static inline unsigned int mbedtls_ssl_session_check_ticket_flags(
+/**
+ * Check if at least one of the given flags is set in
+ * the session ticket. See the definition of
+ * `MBEDTLS_SSL_TLS1_3_TICKET_FLAGS_MASK` to get all
+ * permitted flags.
+ */
+static inline int mbedtls_ssl_session_ticket_has_flags(
```

Comment:
If this function (and friends) are only for TLS 1.3 session tickets, should their names start with ` mbedtls_ssl_tls13_`?

Reply:
I agree, session tickets aren't just TLS 1.3, but the underlying function `mbedtls_ssl_session_get_ticket_flags()` masks with `MBEDTLS_SSL_TLS1_3_TICKET_FLAGS_MASK`.

I don't (yet) know the code well enough to know if this means we're only looking at TLS 1.3 here, hence my question.

But if we're only looking at "TLS 1.3 session tickets", I think we should include `tls13` in the function name.

And if we're not only looking at "TLS 1.3 session tickets", then `MBEDTLS_SSL_TLS1_3_TICKET_FLAGS_MASK` might be wrongly named.

@ronald-cron-arm ?

Reply:
I think that `mbedtls_ssl_session_get_ticket_flags()`,`mbedtls_ssl_session_ticket_has_flags()`, `mbedtls_ssl_session_set_ticket_flags()` and `mbedtls_ssl_session_clear_ticket_flags()` are generic flag handling functions and thus I am fine if `tls13` is not in the function name. In the future their code could evolve into something like:
```
if (session->tls_version == MBEDTLS_SSL_VERSION_TLS1_3) {
    ... use TLS 1.3 flags
}
...
```
`mbedtls_ssl_session_ticket_allow_psk()` and `mbedtls_ssl_session_ticket_allow_psk_ephemeral()` are not generic flag handling functions but generic in the TLS ticket context thus I think no `tls13` in the name is fine as well.

One other point, those functions are internal, thus we can change them at any time.

Reply:
In the above I was rather thinking about the future and an hypothetical TLS 1.4 but TLS 1.2 is more concrete. TLS 1.2 defines tickets but not ticket flags thus if both TLS 1.2 and TLS 1.3 are enabled at build time, one could be tempted to use them in a TLS 1.2 context. It shouldn't be too bad but it isn't worth investigating that. Adding `tls13` in the function names is safer.